File: src/main/java/forpleuvoir/hiirosakura/client/feature/task/executor/base/ExecutorParser.java
package forpleuvoir.hiirosakura.client.feature.task.executor.base;

import forpleuvoir.hiirosakura.client.feature.event.base.Event;
import forpleuvoir.hiirosakura.client.feature.task.executor.*;
import org.jetbrains.annotations.Nullable;

/**
 * 执行器解析器
 *
 * @author forpleuvoir
 * <p>#project_name hiirosakura
 * <p>#package forpleuvoir.hiirosakura.client.feature.task.executor.base
 * <p>#class_name ExecutorParser
 * <p>#create_time 2021-07-23 16:14
 */
public class ExecutorParser {

    public static IExecutor parse(String script, @Nullable Event event) {
        return new JavaScriptExecutor(script, event);
    }
}

File: src/main/kotlin/forpleuvoir/hiirosakura/client/feature/task/executor/base/ExecutorParser.kt
package forpleuvoir.hiirosakura.client.feature.task.executor.base

import forpleuvoir.hiirosakura.client.feature.event.base.Event
import forpleuvoir.hiirosakura.client.feature.task.executor.JavaScriptExecutor

/**
 * 执行器解析器
 *
 * @author forpleuvoir
 *
 * #project_name hiirosakura
 *
 * #package forpleuvoir.hiirosakura.client.feature.task.executor.base
 *
 * #class_name ExecutorParser
 *
 * #create_time 2021-07-23 16:14
 */
object ExecutorParser {
	fun parse(script: String, event: Event? = null): IExecutor {
		return JavaScriptExecutor(script, event)
	}
}